src/create-extension-pack.js

```diff
@@ -0,0 +1,184 @@
+    console.log('Generated ' + PACKAGE_JSON + ' file at: ' + packJsonPath);
+    fs.writeFileSync(licensePath, genLicense());
+    fs.writeFileSync(readmePath, genReadme());
+
+    await vsce.createVSIX({
+        'cwd': packFolderPath,
+        'packagePath': dist(),
+        'useYarn': true
+    });
+
+    async function resolveExtensions() {
+        const extensionsArr = [];
+        for (const extension of fs.readdirSync(extensions())) {
+            const extDataPath = extensions(extension, PACKAGE_JSON);
+            if (!fs.existsSync(extDataPath)) {
+                console.log('No ' + PACKAGE_JSON + ' found for: ' + extension);
+                continue;
+            }
+
+            const content = fs.readFileSync(extDataPath, 'utf-8');
+            const extData = JSON.parse(content);
+
+            const extVsixPath = dist(extData.name + '-' + packVersion + '.vsix');
+            const available = await isAvailable(extVsixPath, extData.name, packVersion);
+            if (!available) {
+                console.log("Skipping extension, i.e. .vsix is not found and " +
+                    "neither published in the registry : " + extVsixPath);
+                continue;
+            }
+
+            const extensionId = extData.publisher + '.' + extData.name;
+            console.log('Adding: ' + extensionId);
+            extensionsArr.push(extensionId);
+        }
+        return Promise.resolve(extensionsArr);
+    }
+
+    async function isAvailable(extVsixPath, extensionName, extensionVersion, namespace = 'vscode') {
+        if (fs.existsSync(extVsixPath)) {
+            return Promise.resolve(true);
+        }
+
+        return isPublished(extensionVersion, extensionName, namespace);
+    }
+})();
+
+function genLicense() {
+    const date = new Date();
+    const year = date.getFullYear();
+    return `MIT License
+
+Copyright(c) ${year} - Ericsson
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files(the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions: 
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.`;
+}
+
+function genReadme(ext) {
```

Comment:
We can be explicit in the naming:

```suggestion
function generateReadme(ext) {
```

Reply:
Done!